lib/bot.js

```diff
@@ -137,10 +137,12 @@ class Bot {
     logger.debug('Channel Mapping', channelName, this.channelMapping[channelName]);
     if (ircChannel) {
       const username = author.username;
+      let nickname = author.username;
       const fromServer = message.server;
-      const userDetails = fromServer.detailsOfUser(author);
```

Comment:
The code was a lot cleaner before this commit. The tests failed because the stub messages don't include the server property. If this is always present in real world situations it's better to change the tests to include it. The updates needed are in [this file](https://github.com/reactiflux/discord-irc/blob/master/test/bot.test.js#L73-L85), wherever `sendToIRC` is called.

You would need to create a stub server, so that you can stub out the `detailsOfUser` method. Since we don't need to pass a constructor anywhere, this is simpler than when stubbing out discord.js's client itself.
Here's an example of how it could look:

```js
const createServerStub = nickname => (
  sandbox.stub().resolves({ 
    detailsOfUser() {
      return { nick: nickname };
    }
  })
);

// Later, in the test itself:
const nickname = 'nick';
const server = createServerStub(nickname);
const message = {
  // other message stuff
  server
};

server.detailsOfUser.should.have.been.calledWith(nickname);
```